lib/requester/requester.js

```diff
@@ -51,97 +54,124 @@ var requests = require('./request-wrapper'),
-     * to the same host are made over the same underlying TCP connection.
-     * @param {CookieJar} [options.cookieJar] A cookie jar to use with Node requests.
-     * @param {Boolean} [options.strictSSL]
-     * @param {Boolean} [options.followRedirects=true] If false, returns a 301/302 as the response code instead of following the redirect
-     * @note `options.keepAlive` is only supported in Node.
-     * @note `options.cookieJar` is only supported in Node.
-     * @constructor
-     */
-    Requester = function (options) {
-        this.keepAlive = _.get(options, 'keepAlive', true);
-        this.cookieJar = _.get(options, 'cookieJar');
-        this.timeout = _.get(options, 'timeout');
-        this.strictSSL = _.get(options, 'strictSSL');
-        this.sendBodyWithGetRequests = _.get(options, 'sendBodyWithGetRequests', false);
-        this.followRedirects = _.get(options, 'followRedirects', true);
-    };
+    Requester;
 
 /**
- * Perform an HTTP request.
+ * Creates a new Requester, which is used to make HTTP(s) requests.
  *
- * @param item
- * @param cb
- * @param scope
+ * @param options
+ * @param {Boolean} [options.keepAlive=true] Optimizes HTTP connections by keeping them alive, so that new requests
+ * to the same host are made over the same underlying TCP connection.
+ * @param {CookieJar} [options.cookieJar] A cookie jar to use with Node requests.
+ * @param {Boolean} [options.strictSSL]
+ * @param {Boolean} [options.followRedirects=true] If false, returns a 301/302 as the response code instead of following the redirect
+ * @note `options.keepAlive` is only supported in Node.
+ * @note `options.cookieJar` is only supported in Node.
+ * @constructor
  */
-Requester.prototype.request = function (item, cb, scope) {
-    var request = item.request,
-        cookieJar = this.cookieJar,
-        requestOptions = util.getRequestOptions(request, {
-            keepAlive: this.keepAlive,
-            cookieJar: cookieJar,
-            timeout: this.timeout,
-            strictSSL: this.strictSSL,
-            sendBodyWithGetRequests: this.sendBodyWithGetRequests,
-            followRedirects: this.followRedirects
-        }),
-        startTime = Date.now();
-
-    return requests(request, requestOptions, function (err, res, resBody) {
-        if (err) { return cb.call(scope || this, err, undefined, request); }
-
-        var responseString,
-            responseTime,
-            responseJSON,
-            cookies,
-            response;
-
-        // holds the response as a string
-        responseString = ((resBody !== null && resBody !== undefined) && resBody.toString) ?
-            resBody.toString() : resBody;
-        if (responseString === '[object ArrayBuffer]') {
-            responseString = util.arrayBufferToString(resBody);
-        }
+inherits(Requester = function (trace, options) {
+    this.options = {
```

Comment:
Let's avoid constructing a new cookie jar every time, by keeping the current logic in place?